### PR TITLE
add config menus

### PR DESCRIPTION
**Pull request by DersuG**
Components should have some kind of `draw_config_menu()` function. The main `App` will call this to construct the config window.

### CODE REVIEW

File: data_controllers/serial_data_controller.py [hunk 1]
```diff
@@ -17,7 +23,7 @@ def __init__(self) -> None:
     def get_config(self) -> dict[str]:
         return {
             'port_name': self.port_name,                # string like 'COM1', 'tty0', etc.
-            'port_baud_rate': self.port_baud_rate,      # int, like 9600 (in ui, use standard values, may alow custom)
+            'port_baud_rate': self.port_baud_rate,      # int, like 9600 (in ui, use standard values, may allow custom)
```
Comment (PeterHall16) [inline data_controllers/serial_data_controller.py:26]: Might be able to autodetect from radio? No idea tbh

File: data_controllers/serial_data_controller.py [hunk 2]
```diff
@@ -45,6 +51,66 @@ def set_config(self, config: dict[str]) -> None:
...
+            elif data == '8 bits':
+                self.port_byte_size = serial.EIGHTBITS
+
+        with gui.group(horizontal=True):
+            gui.add_text('Port:')
+            gui.add_combo(items=self.available_ports, width=-1, tag=SerialDataController.TAG_CONFIG_MENU_PORT_COMBO, callback=set_port_name)
+        gui.add_button(label='Rescan ports', callback=rescan_ports)
+        with gui.group(horizontal=True):
+            gui.add_text('Baud rate:')
+            gui.add_input_int(default_value=9600, width=-1, callback=set_port_baud_rate)
```
Comment (PeterHall16) [inline data_controllers/serial_data_controller.py:101]: Don't love hardcoding in baud rate like this maybe we use a global variable? Just don't want to accidentally have different values in the code if we decide to change it later on

**Review by PeterHall16 — APPROVED** [review]
Code looks solid - also tested the UI on my Mac and seems to work properly.

File: data_controllers/serial_data_controller.py (hunk 2 above)
Comment (DersuG) [reply data_controllers/serial_data_controller.py:101]: this is just the default, and anything that needs this value should be able to get it from the data controller instance. In the future i want to have  the config menu save values you select, so when the program runs the next time they're already set.

Comment (PeterHall16) [reply data_controllers/serial_data_controller.py:101]: Understood, no changes needed then, ready to merge

File: data_controllers/serial_data_controller.py (hunk 1 above)
Comment (DersuG) [reply data_controllers/serial_data_controller.py:26]: looks like this is possible

**Pull request merged**